tools/schemapi/vega_expr.py

```diff
@@ -0,0 +1,964 @@
+            f"doc={self.doc!r}\n"
+            ")"
+        )
+
+
+@dataclasses.dataclass
+class VegaExprParam:
+    name: str
+    required: bool
+    variadic: bool = False
+
+    @staticmethod
+    def star_args() -> LiteralString:
+        return f"{STAR_ARGS}: Any"
+
+    def to_str(self) -> str:
+        """Return as an annotated parameter, with a default if needed."""
+        if self.required:
+            return f"{self.name}: {INPUT_ANNOTATION}"
+        elif not self.variadic:
+            return f"{self.name}: {INPUT_ANNOTATION} = {NONE}"
+        else:
+            return self.star_args()
+
+    @classmethod
+    def iter_params(cls, raw_texts: Iterable[str], /) -> Iterator[Self]:
+        """Yields an ordered parameter list."""
+        is_required: bool = True
+        for s in raw_texts:
+            if s not in {OPEN_PAREN, CLOSE_PAREN}:
+                if s == OPEN_BRACKET:
+                    is_required = False
+                    continue
+                elif s == CLOSE_BRACKET:
+                    is_required = True
+                    continue
+                elif s.isalnum():
+                    yield cls(s, required=is_required)
+                elif s == ELLIPSIS:
+                    yield cls(STAR_ARGS, required=False, variadic=True)
+                else:
+                    continue
+
+
+def _parse_expressions(url: str, /) -> Iterator[VegaExprNode]:
+    """Download, read markdown and iteratively parse into signature representations."""
+    for tok in read_ast_tokens(download_expressions_md(url)):
+        if (
+            (children := tok.get(CHILDREN)) is not None
+            and (child := next(iter(children)).get(RAW)) is not None
+            and (match := FUNCTION_DEF_LINE.match(child))
+        ):
+            node = VegaExprNode(match[1], children)
+            if node.is_callable():
+                yield node.with_parameters().with_signature()
+    request.urlcleanup()
+
+
+def parse_expressions(url: str, /) -> Iterator[VegaExprNode]:
+    """
+    Eagerly parse signatures of relevant definitions, then yield with docs.
+
+    Ensures each doc can use all remapped names, regardless of the order they appear.
+    """
+    eager = tuple(_parse_expressions(url))
+    VegaExprNode.remap_title.refresh()
+    for node in eager:
+        yield node.with_doc()
+
+
+EXPR_MODULE_PRE = '''\
+"""Tools for creating transform & filter expressions with a python syntax."""
+
+from __future__ import annotations
+
+import sys
+from typing import Any, TYPE_CHECKING
+
+from altair.expr.core import {const}, {func}
+from altair.vegalite.v5.schema.core import ExprRef as _ExprRef
+
+if sys.version_info >= (3, 12):
+    from typing import override
+else:
+    from typing_extensions import override
+
+if TYPE_CHECKING:
+    from altair.expr.core import {return_ann}, {input_ann}
+
+
+class {metaclass}(type):
+    """Metaclass providing read-only class properties for :class:`expr`."""
+
+    @property
```

Comment:
Update summary to state that properties are a **feature** of the metaclass.

Main point is that it is the metaclass for `expr`